designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/RemoteEnvironment.scala

```diff
@@ -106,44 +108,24 @@ trait StandardRemoteEnvironment extends FailFastCirceSupport with RemoteEnvironm
     (for {
       validation <- EitherT(validateProcess(localProcess))
       _ <- EitherT.fromEither[Future](if (validation.errors != ValidationErrors.success) Left[EspError, Unit](MigrationValidationError(validation.errors)) else Right(()))
-      processExistsOnRemote <- checkIfProcessExistsOnRemote(localProcess)
-      _ <- if (processExistsOnRemote) {
-        isArchivedOnRemote(localProcess)
-      } else {
-        createRemoteProcess(localProcess, category)
+      processEither <- fetchProcessDetails(localProcess.id)
+      _ <- processEither match {
+        case Right(processDetails) if processDetails.isArchived => EitherT.leftT[Future, EspError](MigrationToArchivedError(processDetails.idWithName.name, environmentId))
+        case Right(_) => EitherT.rightT[Future, EspError](())
+        case Left(_: RemoteEnvironmentCommunicationError) => createProcessOnRemote(localProcess, category)
```

Comment:
You must check:
`Left(RemoteEnvironmentCommunicationError(StatusCodes.NotFound, _))`, currently you handles 400, 5xx, etc as well

Reply:
u r right, missed it, thanks